### ISSUE TITLE
Loki: Logs Disappear from Logs volume:

### ISSUE THREAD

**Issue by TelloDMan** [grafana/loki#13135]
# **Grafana loki Logs disappear and re-appear after restart**
I currently have a huge amount of syslogs being stored in grafana loki , which are being sent using fluentbit. I was not having problems until i faced an issue where the logs appeared to be disappearing from the logs volume:
![disappear](https://github.com/grafana/loki/assets/80420517/51b5ff57-363e-419d-872e-b5ac8af37efb)

I knew logs are being sent and stored since i was running fluentbit in debug mode and no logs were being generated on grafana loki. So i restarted the docker container for loki. and re-ran the query and logs re-appeared.
![appear](https://github.com/grafana/loki/assets/80420517/da198ec1-8df9-40e3-ae34-adff4ad61013)

this issue is becomming common almost everyday , idk what is causing the issue, i can provide my configs 

My Loki Config (ver:2.9)
```
auth_enabled: false


server:
  http_listen_port: 3100
  grpc_server_max_recv_msg_size: 12667131  
  grpc_server_max_send_msg_size: 12667131    
  http_server_read_timeout: 10m
  http_server_write_timeout: 10m

common:
  ring:
    instance_addr: 127.0.0.1
    kvstore:
      store: inmemory
  replication_factor: 1
  path_prefix: /loki

limits_config:
  enforce_metric_name: false
  reject_old_samples: false
  reject_old_samples_max_age: 168h
  ingestion_rate_mb: 10
  ingestion_burst_size_mb: 20
  ingestion_rate_strategy: global
  split_queries_by_interval: 30m
  retention_period: 17532h
  
schema_config:
  configs:
    - from: 2020-10-24
      store: tsdb
      object_store: filesystem
      schema: v13
      index:
        prefix: index_
        period: 24h

querier:
  query_timeout: 10m
  engine:
    timeout: 5m


query_range:
  align_queries_with_step: true
  max_retries: 5
  cache_results: true

  results_cache:
    cache:
      enable_fifocache: true
      fifocache:
        max_size_items: 1024
        validity: 24h

      
      
storage_config:
  filesystem:
    directory: /loki/chunks 

ruler:
  alertmanager_url: http://localhost:9093


analytics:
  reporting_enabled: false
```

**To Reproduce**
1. Started Loki (ver:2.9)
2. Made A couple of large queries , waited for couple of days 
3. Made the same large query again , now appeared logs are disappeared
4.Restarted the docker container containing loki, made the same query and now logs appeared again

**Expected behavior**
Expected behavior is not to lose logs and no need for the restart if loki is storing the logs.

**Environment:**
 - Infrastructure: Ubuntu 22.04.4 LTS(bare matel)
 - Deployment tool: Docker compose v3.8
Docker Compose Config
```
#Loki for Graphana
  loki:
    container_name: loki
    image: grafana/loki:2.9.0
    user: "0"
    restart: unless-stopped
    ports:
      - "3100:3100"
    command: -config.file=/etc/loki/local-config.yaml
    volumes:
      - "/home/user/log-server/loki/etc/local-config.yaml:/etc/loki/local-config.yaml"
      - "/etc/loki:/loki"
    network_mode: host
```

Is this an issue in my configuration or a bug in loki?

**Comment by teslawomir** [comment]
Any update on this? i am facing same issue..

**Comment by rsaelim** [comment]
I encountered the same issue with Loki 2.9.0. This issue is gone for me after I upgraded to v2.9.8

**Comment by TelloDMan** [comment]
![Image](https://github.com/user-attachments/assets/11daf3be-6fb9-446e-afd4-76ac895557d4)

Yes Dear @teslawomir the issue is on grafana loki 2.9 as stated in the release note the bug was patched in v2.9.1
